system/lib/pthread/library_pthread.c

```diff
@@ -846,13 +848,12 @@ em_queued_call* emscripten_async_waitable_run_in_main_runtime_thread_(
   return q;
 }
 
-void EMSCRIPTEN_KEEPALIVE emscripten_async_queue_on_thread_(
+int EMSCRIPTEN_KEEPALIVE _emscripten_call_on_thread(
+  int forceAsync,
   pthread_t targetThread, EM_FUNC_SIGNATURE sig, void* func_ptr, void* satellite, ...) {
   int numArguments = EM_FUNC_SIG_NUM_FUNC_ARGUMENTS(sig);
   em_queued_call* q = em_queued_call_malloc();
   assert(q);
-  if (!q)
-    return;
```

Comment:
I removed this as it seems incorrect - we have an assertion right above it? With this return we would just silently drop messages when out of memory.

Reply:
Hmm, this does not look good to remove. Asserts will not be present in release builds. The issue here is that if malloc fails, we will be unable to queue the message. When we are in such a constrained memory situation, the only thing we can do is to silently cancel, hoping that whatever code path we have elsewhere that saw the malloc failure, would be able to deal with the OOM situation.

If we do want to change this to something else, it should be something along the lines of
```
if (!q) abortCurrentThread();
```
The original behavior of `if (!q) return;` would cause the calling thread to hang indefinitely, waiting for a message to complete that never will, which is a relatively good way to park the calling thread to a "we have hung now" state.

Reply:
Ok, I restored the old code, and added a TODO.

This pattern appears in a bunch of places in this code and might be worth changing. Seems like for async messages it just drops them silently, not even hanging, atm?

Reply:
Yeah, async messages would be silently dropped. Given that the page is in a tight OOM state, it's likely that something else will soon OOM anyways (if it hasn't already), which is why I did not worry about that too much. But adding an explicit abort here would be ok.